src/index.ts

```diff
@@ -63,21 +63,42 @@ interface DecryptJWEParams {
 }
 
 interface AuthParams {
+  nonce: string
+  aud: string
   paths: Array<string>
 }
 
+const sign = async (
+  payload: Record<string, any>,
+  did: string,
+  secretKey: Uint8Array,
+  protectedHeader?: Record<string, any>
+) => {
+  const kid = `${did}#${did.split(':')[2]}`
+  const signer = NaclSigner(u8a.toString(secretKey, B64))
+  const header = toStableObject(Object.assign(protectedHeader || {}, { kid, alg: 'EdDSA' }))
```

Comment:
Not an issue but we could have instead in the arguments `protectedHeader: Record<string, any> = {}` and then here `toStableObject({ ...protectedHeader, kid, alg: 'EdDSA' })`

Reply:
good point!